### PR TITLE
Town report implemented

### PR DESCRIPTION
**Pull request by Mankur11**

### CODE REVIEW

File: .idea/encodings.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ostapgv) [inline .idea/encodings.xml]: Read about `.gitignore` file purpose and add `.idea` folder there (maybe smth else). You should not commit files which are not related to the project.

File: .idea/misc.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ostapgv) [inline .idea/misc.xml]: remove IDE related files from the pull request

File: index.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title>Section 7: Get Ready for the Future: ES6 / ES2015</title>
+        
+        <style>
+            /*.box {*/
```
Comment (ostapgv) [inline index.html]: Do you need commented styles here? If you want to add styles, don't inline them, import separate css file instead.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            /*.green { background-color: green; }*/
+            /*.blue { background-color: dodgerblue; }*/
+            /*.orange { background-color: orangered; }*/
+            
+        </style>
+        
+    </head>
+
+    <body>
+        <!--<h1>Section 7: Get Ready for the Future: ES6 / ES2015</h1>-->
```
Comment (ostapgv) [inline index.html]: Don't left commented code blocks in the PR.

File: index.html [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        <!---->
+        <!--<div class="box green">I'm green!</div>-->
+        <!--<div class="box blue">I'm blue!</div>-->
+        <!--<div class="box orange">I'm orange!</div>-->
+        <!---->
+        <!--<script src="script.js"></script>-->
+    </body>
```
Comment (ostapgv) [inline index.html]: Your js file is not imported anywhere.

File: script.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        this.name = name;
+        this.area = area;
+        this.age = age;
+        this.numOfTrees = numOfTrees;
+    }
+
+    parkDensity(){
+        console.log(`${this.name} has a tree density of ${this.numOfTrees/this.area} trees per square km`);
+    }
+
```
Comment (ostapgv) [inline script.js]: Remove empty line here.

File: .idea/encodings.xml (hunk 1 above)
Comment (Mankur11) [reply .idea/encodings.xml]: README updated, created .gitignore file

File: index.html (hunk 3 above)
Comment (Mankur11) [reply index.html]: I don't need any styles here at all. Removed.

File: index.html (hunk 4 above)
Comment (Mankur11) [reply index.html]: Cleared

File: index.html (hunk 5 above)
Comment (Mankur11) [reply index.html]: fixed

File: script.js (hunk 6 above)
Comment (Mankur11) [reply script.js]: removed

File: .idea/encodings.xml (hunk 1 above)
Comment (ostapgv) [reply .idea/encodings.xml]: At first remove `.idea` folder from the branch and commit. Then add `.gitignore` file, ignore `.idea` folder and commit again. And please take a look at your commits before pushing them to detect if everything looks ok there and there's no unexpected unnecessary files presented.

File: script.js [hunk 7]
```diff
@@ -13,13 +13,11 @@ class ParksCalculator {
     }
```
Comment (ostapgv) [inline script.js]: 7 row: rename `parksNamesArray` to `getParkNamesArray` or `getParkNames`. Method should starts from the verb.

Comment (ostapgv) [inline script.js]: 11 row: Do you need the `getParks` method? It's just returning the result of another method.

File: script.js [hunk 8]
```diff
@@ -13,13 +13,11 @@ class ParksCalculator {
...
 
     generateParkByName(parkName) {
-        let parkSize = ['tiny', 'small', 'normal', 'big', 'huge'];
         return ({
             name: parkName,
             age: Math.floor(Math.random() * 300),
             area: Math.floor(Math.random() * 5000),
```
Comment (ostapgv) [inline script.js]: The `area` of the park cannot be equal 0.

File: script.js [hunk 9]
```diff
@@ -13,13 +13,11 @@ class ParksCalculator {
...
-            numOfTrees: Math.floor(Math.random() * 1500),
-            size: parkSize[Math.floor(Math.random() * 5)]
+            numOfTrees: Math.floor(Math.random() * 1500)
         });
     }
```
Comment (ostapgv) [inline script.js]: 30, 38 rows: Method names should start from the verb. you don't need the `parksArray` param here, use `this.parks` instead.

Comment (ostapgv) [inline script.js]: row 32: `Array.prototype.reduce` is perfectly fit here instead of `for`. If you use for, try to avoid such abstract variable names like `value`. It could look like `for (let park of this.parks) {` or `this.parks.forEach((park) => {`.

File: script.js [hunk 10]
```diff
@@ -54,6 +52,56 @@ parksCalculator.averageParksAge(parksArray);
 parksCalculator.parksDensity(parksArray);
```
Comment (ostapgv) [inline script.js]: 43 row: you don't need the `parksArray`. `const parksCalculator = new ParksCalculator();` is enough.

Comment (ostapgv) [inline script.js]: Split you methods  to two parts:
- first one returns the value we expect to get
- second one logs to the console
There's really rare case when the function with some calculations just logs calculated value.

Example: 
split `averageParksAge` to `getAverageParksAge` and `logAverageParksAge`.

File: script.js [hunk 11]
```diff
@@ -54,6 +52,56 @@ parksCalculator.averageParksAge(parksArray);
...
 
 
+//Streets report
+class StreetsCalculator {
```
Comment (ostapgv) [inline script.js]: Fix the all mentioned above for the `StreetsCalculator` class as well.

File: script.js (hunk 7 above)
Comment (Mankur11) [reply script.js]: done

Comment (Mankur11) [reply script.js]: no, I don't. getParks method deleted

File: script.js (hunk 8 above)
Comment (Mankur11) [reply script.js]: added 1 to area's Math.random()

File: script.js (hunk 10 above)
Comment (Mankur11) [reply script.js]: done.

Comment (Mankur11) [reply script.js]: fixed

File: script.js (hunk 9 above)
Comment (Mankur11) [reply script.js]: return this.parks.reduce((accumulator, currentValue) => accumulator + currentValue.age); - something wrong here, don't understand yet

Comment (Mankur11) [reply script.js]: fixed

File: script.js [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+
+    logAverageParksAge() {
+        console.log(`--Parks report--\n\nOur ${this.parks.length} parks have an average age of ${this.getAverageParksAge() / this.parks.length} years`);
+    }
+
+    getParksDensity(park) {
+        return park.numOfTrees / park.area
+    }
+
+    isThousandTrees() {
```
Comment (ostapgv) [inline script.js]: isThousandTrees -> hasParkWithThousandTrees

File: script.js [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+    getAverageParksAge() {
+        return this.parks.reduce((accumulator, currentValue) => accumulator + currentValue.age);
+    }
+
+    logAverageParksAge() {
+        console.log(`--Parks report--\n\nOur ${this.parks.length} parks have an average age of ${this.getAverageParksAge() / this.parks.length} years`);
+    }
+
+    getParksDensity(park) {
```
Comment (ostapgv) [inline script.js]: getParksDensity -> getParkDensity

File: script.js [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+            }
+        }
+    }
+
+    logParksDensity() {
+        for (let park of this.parks) {
+            console.log(`${park.name} has a tree density of ${this.getParksDensity(park)} trees per square km`);
+        }
+        if (this.isThousandTrees()) {
+            console.log(`${this.isThousandTrees()} has more than 1000 trees`);
```
Comment (ostapgv) [inline script.js]: This will print `true/false has more than 1000 trees`

File: script.js [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+        for (let park of this.parks) {
+            if (park.numOfTrees > 1000) {
+                return park.name;
+            } else {
+                return false;
+            }
+        }
+    }
+
+    logParksDensity() {
```
Comment (ostapgv) [inline script.js]: Rewrite this method to be accurate in what you want to log. Do you want to log the info about each park that has more than 1000 trees or the info that there is a park with more than 1000 trees?

File: README.md [hunk 16]
```diff
@@ -1 +1,17 @@
...
+There are only 3 arks and 4 streets in town.
+All parks and streets have a name and a build year.
+
+We need end-of-year report with the following:
+
+1. Tree density of each park in the town
+ (number of trees/park area)
+2. Average age of each town's park
+ (sum of all ages/number of parks)
+3. The name of the parks that has more than 1000 trees
```
Comment (ostapgv) [inline README.md]: it's not done

File: script.js (hunk 9 above)
Comment (Mankur11) [reply script.js]: fixed

File: script.js (hunk 11 above)
Comment (Mankur11) [reply script.js]: fixed

File: script.js [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    generateParks(parksNames) {
+        return parksNames.map((parkName) => {
+            return this.generateParkByName(parkName);
+        })
+    }
+
+    getAverageParksAge() {
+        return this.parks.reduce((accumulator, currentValue) => accumulator + currentValue.age);
```
[the remark this answers is not in the document]
Comment (Mankur11) [reply script.js]: fixed

File: script.js [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    logAverageParksAge() {
+        console.log(`--Parks report--\n\nOur ${this.parks.length} parks have an average age of ${this.getAverageParksAge() / this.parks.length} years`);
+    }
+
+    getParksDensity(park) {
+        return park.numOfTrees / park.area
```
[the remark this answers is not in the document]
Comment (Mankur11) [reply script.js]: need 2 modify my prettier in WS, fixed ; manually

File: script.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (Mankur11) [reply script.js]: fixed

File: script.js [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+            return this.generateParkByName(parkName);
+        })
+    }
+
+    getAverageParksAge() {
+        return this.parks.reduce((accumulator, currentValue) => accumulator + currentValue.age);
+    }
+
+    logAverageParksAge() {
+        console.log(`--Parks report--\n\nOur ${this.parks.length} parks have an average age of ${this.getAverageParksAge() / this.parks.length} years`);
```
[the remark this answers is not in the document]
Comment (Mankur11) [reply script.js]: fixed. But in Streets report it is a different situation

File: script.js (hunk 12 above)
Comment (Mankur11) [reply script.js]: fixed. I've tried

File: script.js (hunk 13 above)
Comment (Mankur11) [reply script.js]: fixed.yes, it's about 1 park

File: script.js (hunk 14 above)
Comment (Mankur11) [reply script.js]: fixed

File: script.js (hunk 15 above)
Comment (Mankur11) [reply script.js]: created logParkWithThousandTrees method for logging about 1000 trees, now logParksDensity is only about density
